From 2246925e3cb16183e25d4e2cfd13fb800df86270 Mon Sep 17 00:00:00 2001
From: Beniamin Sandu <beniaminsandu@gmail.com>
Date: Sun, 25 Jun 2023 19:58:08 +0300
Subject: [PATCH] aesce: do not specify an arch version when enabling crypto
 instructions

Building mbedtls with different aarch64 tuning variations revealed
that we should use the crypto extensions without forcing a particular
architecture version or core, as that can create issues.

Upstream-Status: Submitted [https://github.com/Mbed-TLS/mbedtls/pull/7834]

Signed-off-by: Beniamin Sandu <beniaminsandu@gmail.com>
---
 library/aesce.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/library/aesce.c b/library/aesce.c
index fe056dc4c..843de3973 100644
--- a/library/aesce.c
+++ b/library/aesce.c
@@ -60,7 +60,7 @@
 #           error "A more recent GCC is required for MBEDTLS_AESCE_C"
 #       endif
 #       pragma GCC push_options
-#       pragma GCC target ("arch=armv8-a+crypto")
+#       pragma GCC target ("+crypto")
 #       define MBEDTLS_POP_TARGET_PRAGMA
 #   else
 #       error "Only GCC and Clang supported for MBEDTLS_AESCE_C"
-- 
2.25.1

